test/index.js

```diff
@@ -977,6 +977,24 @@ describe('def', function() {
     eq(dec(new Number(-1)), -2);
   });
 
+  it('supports the "RegexFlags" type', function() {
+    eq($.RegexFlags.name, 'sanctuary-def/RegexFlags');
+    eq($.RegexFlags.test(''), true);
+    eq($.RegexFlags.test('g'), true);
+    eq($.RegexFlags.test('i'), true);
+    eq($.RegexFlags.test('m'), true);
+    eq($.RegexFlags.test('gi'), true);
+    eq($.RegexFlags.test('gm'), true);
+    eq($.RegexFlags.test('im'), true);
+    eq($.RegexFlags.test('gim'), true);
+    //  String objects are not acceptable.
+    eq($.RegexFlags.test(new String('')), false);
+    //  Flags must be alphabetically ordered.
+    eq($.RegexFlags.test('mg'), false);
```

Comment:
Preferable, in my view. I see `RegexFlags` as a union type:

```haskell
data RegexFlags = Empty | G | I | M | GI | GM | IM | GIM
```

Strings happen to be the best way to achieve this in JavaScript, but I'd like us to think of `'gm'` as we think of `GM` in the Haskell definition above. `MG` is not a constructor in the Haskell definition; similarly `'mg'` is not a member of `RegexFlags`.

Do you find this argument compelling?